cmake_modules/commonSetup.cmake

```diff
@@ -378,6 +384,22 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
            add_definitions(/J)
         endif ()
     endif ()
+
+
+    #===  WARNING ======
+    # Temporary disable warnings. Reenable them one by one to reexamine and fix them.
+    IF (NOT ALL_WARNINGS_ON)
+      set (WARNINGS_IGNORE "/wd4267 /wd4244 /wd6340 /wd6297 /wd4018 /wd4302 /wd4311 /wd4320 /wd4800")     # data conversion warnings
+      set (WARNINGS_IGNORE  "${WARNINGS_IGNORE} /wd4251 /wd4275")      # dll-interface for used by clients
+      set (WARNINGS_IGNORE  "${WARNINGS_IGNORE} /wd6246")   # local variable hidden by outter scope
+      set (WARNINGS_IGNORE  "${WARNINGS_IGNORE} /wd6031")   # Return value ignored
+      set (WARNINGS_IGNORE  "${WARNINGS_IGNORE} /wd4005")   # MACRO redef: same value
+      set (WARNINGS_IGNORE  "${WARNINGS_IGNORE} /wd4503")   # decorated name length exceeded, name was truncated
+      
+      set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${WARNINGS_IGNORE}")
+    ENDIF()
```

Comment:
Are we setting ourselves up for ignoring these forever, or do we have a ticket open to address them?

Reply:
Definitely not forever. Ideally we don't need to disable any warnings, but it just too much to deal with as one bug (over 40K warnings). 

Yes, we will have ticket to address them. We may have one ticket to enable one or few warnings in a manageable way.